File: InJava/app/src/main/java/com/barryzeha/pomodoroapp/common/TimerState.java
package com.barryzeha.pomodoroapp.common;

/****
 * Project PomodoroApp
 * Created by Barry Zea H. on 18/10/2022
 * Copyright (c)  All rights reserved.
 ***/
public enum TimerState {
    NotStarted,
    OnStart,
    OnPause,
    OnStop,
    OnNext,
    OnCompleted,
    OnCreatedTask,
    NoHaveTask,
    CompletedTask,
    CompletedBreak;
}

File: InKotlin/app/src/main/java/com/barryzeha/pomodoroapp/common/TimerState.kt
package com.barryzeha.pomodoroapp.common

/****
 * Project PomodoroApp
 * Created by Barry Zea H. on 05/10/2022
 * Copyright (c)  All rights reserved.
 ***/

enum class TimerState {

    NotStarted,
    OnStart,
    OnPause,
    OnStop,
    OnNext,
    OnCompleted,
    OnCreatedTask,
    NoHaveTask,
    CompletedTask,
    CompletedBreak;

    companion object {
        fun toMyEnum(myEnumString:String):TimerState{
            try{
                return valueOf(myEnumString)
            }
            catch (e:Exception){
                return OnStop
            }
        }
    }


}